tests/flows/jobs/billing_journal/line_processors/test_base_processor.py

```diff
@@ -0,0 +1,93 @@
+from decimal import Decimal
+from typing import ClassVar
+
+from swo_aws_extension.constants import ItemSkusEnum, UsageMetricTypeEnum
+from swo_aws_extension.flows.jobs.billing_journal.line_processors.base_processor import (
+    GenerateItemJournalLines,
+)
+
+
+class _ValidatorAlwaysTrue:
+    def validate(self, discount, amount, service_name, account_metrics, tolerance_rate):
+        return True
+
+
+class _ValidatorAlwaysFalse:
+    def validate(self, discount, amount, service_name, account_metrics, tolerance_rate):
+        return False
+
+
+class _DummyProcessor(GenerateItemJournalLines):
+    _item_sku = ItemSkusEnum.AWS_USAGE.value
+    _metric_id = UsageMetricTypeEnum.USAGE.value
+    _validator = _ValidatorAlwaysTrue
+    _exclude_services: ClassVar[list] = []
+
+
+class _DummyProcessorWithAffixes(GenerateItemJournalLines):
+    _item_sku = ItemSkusEnum.AWS_USAGE.value
+    _metric_id = UsageMetricTypeEnum.USAGE.value
+    _validator = _ValidatorAlwaysTrue
+    _prefix = "PRE- "
+    _suffix = " - SUF"
+    _exclude_services: ClassVar[list] = []
+
+
+class _DummyProcessorWithCombos(GenerateItemJournalLines):
+    _item_sku = ItemSkusEnum.AWS_USAGE.value
+    _metric_id = UsageMetricTypeEnum.USAGE.value
+    _validator = _ValidatorAlwaysTrue
+    _dynamic_exclude_services: ClassVar[list] = [UsageMetricTypeEnum.MARKETPLACE.value]
+    _exclude_services: ClassVar[list] = ["Excluded Service"]
+
+
+def test_can_process_returns_expected():
+    proc = _DummyProcessor(billing_discount_tolerance_rate=1, discount=0)
+    assert proc.can_process(ItemSkusEnum.AWS_USAGE.value)
+    assert not proc.can_process(ItemSkusEnum.AWS_MARKETPLACE.value)
```

Comment:
It'd be great if we follow the AAA pattern as defined in our guides. Each section should be separated by a blank line, and we shouldn't add blank lines within each group. 

This way, we can see the test should be split into two or we should use the parametrize, since we cannot have two Arrange blocks.


```python
@pytest.mark.parametrize(("metric_id", "expected_result"), [
    (ItemSkusEnum.AWS_USAGE.value, True),
    (ItemSkusEnum.AWS_MARKETPLACE.value, False),
])
def test_can_process_returns_expected(metric_id, expected_result):
    proc = _DummyProcessor(billing_discount_tolerance_rate=1, discount=0)
    
    result =  proc.can_process(metric_id)
    
    assert result is expected_result
```

Please, take a look at all the test and let me know if I can help with anything